test/compressor_test.go

```diff
@@ -785,3 +785,70 @@ func (s) TestGzipBadChecksum(t *testing.T) {
 		t.Errorf("ss.Client.UnaryCall(_) = _, %v\n\twant: _, status(codes.Internal, contains %q)", err, gzip.ErrChecksum)
 	}
 }
+
+// fakeCompressor returns a messages of a configured size, irrespective of the
+// input.
+type fakeCompressor struct {
+	decompressedMessageSize int
+}
+
+func (f *fakeCompressor) Compress(w io.Writer) (io.WriteCloser, error) {
+	return nopWriteCloser{w}, nil
+}
+
+func (f *fakeCompressor) Decompress(io.Reader) (io.Reader, error) {
+	return bytes.NewReader(make([]byte, f.decompressedMessageSize)), nil
+}
+
+func (f *fakeCompressor) Name() string {
+	// Use the name of an existing compressor to avoid interactions with other
+	// tests since compressors can't be un-registered.
+	return "gzip"
```

Comment:
Can we use `t.Name` as the name for the compressor and thereby avoid interactions with other tests?

Reply:
I tried this earlier, but the clients in the following test pick up the registered compressor and start failing.
https://github.com/grpc/grpc-go/blob/e8055ea11f96238b0fb14b1062abb598c33fc9c3/test/compressor_test.go#L617

There's no way to unregister a compressor presently. There is a deprecated `WithCompressor` that allows using a compressor without registration that I can use.

Reply:
Do you mind filing an issue for us to think about how we can handle this. It is not ideal for one test to influence every test that comes after it.

Reply:
Opened https://github.com/grpc/grpc-go/issues/7960.